### DISCUSSION TITLE
TabbedContent: How to add new tabs programatically?

### DISCUSSION THREAD

**Discussion by guedes** [Textualize/textual#2668] [Q&A] [answered]
Actually I only know how to create new tabs in the "compose" phase and I'd like to add more tabs while program is running.  I couldn't find examples how to do. I fell that I'm missing something obvious.

Is it possíble?

**Comment by davep** [comment]
Currently there isn't a [public interface](https://textual.textualize.io/widgets/tabbed_content/) for adding more tabbed content after it's been composed.

**Reply by guedes** [reply]
Thanks for reply and sorry for opening the issue in the wrong place.
 
So, should I use `Tabs` instead of `TabbedContent`  if I want to add more Tabs? There are plans to add a public interface for that?

**Reply by davep** [reply]
No need to apologise, wasn't the wrong place, I just moved it to where it might be better for others to find longer-term. 

Yes, for now, I'm afraid dropping to Tabs is likely the way to go if you want the tab set to be more dynamic. 

I don't think there's anything on the board for extending the tabbed content widget in this way, but I think there should be. I'll add an issue as a reminder for us to review this.

**Comment by guedes** [comment, marked as answer]
Hi @davep ! 

It seems that this PR https://github.com/Textualize/textual/pull/2751 made this possible, right? Should I close this discussion?

**Reply by davep** [reply]
Feel free to close if you wish, but I think the best approach here is to change what's marked as the correct answer (which I just did). ;-)

**Reply by guedes** [reply]
Nice! Thanks!
